### PR TITLE
Doc 08 mediation local fdr

### PR DESCRIPTION
**Pull request by TomAugspurger**
Closes https://github.com/statsmodels/statsmodels/issues/3440

The description for Mediation came from Kirby's notebook [here](http://nbviewer.jupyter.org/urls/umich.box.com/shared/static/jpmd9y99259u6dv0rj6p46993981m7zm.ipynb)

Anything else to be added?

### CODE REVIEW

File: docs/source/stats.rst [hunk 1]
```diff
@@ -210,6 +210,13 @@ to one way ANOVA, but still in developement
 
    pairwise_tukeyhsd
 
+.. currentmuodule:: statsmodels.stats.multitest
+
+.. autosummary::
+   :toctree: generated/
+
+   local_fdr
+
```
Comment (josef-pkt) [inline docs/source/stats.rst]: `NullDistribution`  needs to be also here AFAICS

Comment (TomAugspurger) [inline docs/source/stats.rst]: And `fdrcorrection_twostage` maybe? `fdrcorrection` is in the above toctree as `fdrcorrection0`, but I don't see two stage.

**Comment by josef-pkt** [comment]
typo `raise RunTimeError`  in stats/multitest.py line 594  (pydev is red when looking at the module)
`RuntimeError`

in stats/api.py needs two more 
`from .multitest import (multipletests, fdrcorrection, fdrcorrection_twostage,
                                     local_fdr, NullDistribution)`

and I think at the end a new
`from .mediation import Mediation`

I didn't see any other missing items, but only spot checked a few more functions.

File: docs/source/stats.rst [hunk 2]
```diff
@@ -216,6 +216,8 @@ to one way ANOVA, but still in developement
    :toctree: generated/
 
    local_fdr
+   fdrcorrection_twostage
+   NullDistribution
```
Comment (josef-pkt) [inline docs/source/stats.rst]:
```
multipletests
fdrcorrection
fdrcorrection_twostage
local_fdr
NullDistribution
```

and delete 
multipletests
fdrcorrection0
on lines 199, 200
those two sandbox imports were backwards compatibility imports and outdated

**Comment by josef-pkt** [comment]
some of the multitest functions in the stats.rst didn't reflect an old refactoring

**Comment by TomAugspurger** [comment]
We can do additional doc fixes after tagging 0.8.0 btw (but this should go in before because that RuntimeError warning.)

**Comment by josef-pkt** [comment]
the api are also important in general because they determine the import path for users, although in this case both multitest and mediation look like public modules

**Comment by TomAugspurger** [comment]
> the api are also important in general because they determine the import path for users

Indeed. I think my last commit got all those.

**Pull request merged**